### ISSUE TITLE
Memory Corruption in WPF DotNet 5 x86

### ISSUE THREAD

**Issue by juanpfry** [dotnet/runtime#51350]
### Description
We have a X86 WPF desktop application, we upgraded to DotNet 5 (from Framework 4.6) and we started experiencing some memory corruption.
Between 15 to 30 minutes of running, different and random systems misbehaved with System.ExecutionEngineException, System.AccessViolationException, or others which after investigation shown a clear heap or a stack corruption. It is hard to rule which one is the root. 

We tried to isolate it for many weeks until we were able to reduce to a small repro.
In Visual Studio, Version 16.9.3, Create new Project: C# WPF Application - Dot Net 5.0, create a new platform config to X86.
Add this code to the MainWindow and execute in Debug mode (Of course this code is for the sake to accelerate the exposure of the bug, in our code we do not trigger the GC at all).

```cs

    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
            
            Task.Run(() =>
            {
                while (true)
                {
                    Thread.Sleep(1);
                    GC.Collect();
                    GC.WaitForPendingFinalizers();
                }
            });
  
            Task.Run(() =>
            {
                while (true)
                {
                    Thread.Sleep(1);
                    System.Diagnostics.Debug.WriteLine("test");
                }
            });
        }
    }

```
With this repro I consistently get:
![image](https://user-images.githubusercontent.com/72634857/114947378-e7708900-9e01-11eb-8e35-fc67f3b1e037.png)
And less often:
![image](https://user-images.githubusercontent.com/72634857/114947446-08d17500-9e02-11eb-82c9-fdcc8baf8dce.png)

We tried this repro case in .netFramework and in Console applications and the bug doesn't manifest there.

So our guess is that something in WPF in .Net 5.0 is corrupting the memory at least on x86

### Configuration

* .NET 5.0
* Visual Studio 16.9.3
* Windows 10
* x86
* WPF desktop C# application

### Regression?

 We didn't experience this problem on .netFramework, and I was not able to repro it in .netFramework nor in .Net3.1
 So it seems highly related to .Net5.0

### Other information

We found no workarounds.

Here you can see how the pointer is corrupted.
![image](https://user-images.githubusercontent.com/72634857/114948514-fc4e1c00-9e03-11eb-9219-7fc9db34209d.png)

**Comment by jkotas** [comment]
This was likely fixed by https://github.com/dotnet/runtime/pull/48445 that is included in 5.0.5 update. Could you please the install 5.0.5 serving release from https://dotnet.microsoft.com/download/dotnet/5.0 and check whether it still crashes?

**Comment by juanpfry** [comment]
Yeah that fixed the problem.
Thanks!
